src/api/Feed.js

```diff
@@ -645,12 +654,23 @@ class Feed extends Base {
 
         try {
             const taskLink = await this.createTaskLink(file, task);
-            const taskAssignments: Array<TaskCollabAssignee> = await Promise.all(
-                assignees.map((assignee: SelectorItem) => {
-                    return this.createTaskCollaborator(file, task, assignee);
-                }),
-            );
 
+            /* TODO: change block to process users in parallel and process groups sequentially
+            -check if length of users + groups = assignees (if not throw an error)
+            -filter out the users from assignees
+            -use the result of filter to store the task collaborators in parallel
+            -filter out the groups from assignees
+            -use result of filter to process each group sequentially */
+            const taskAssignments: Array<TaskCollabAssignee> = flatten<TaskCollabAssignee, TaskCollabAssignee>(
+                await Promise.all(
+                    assignees.map((assignee: SelectorItem): Promise<Array<TaskCollabAssignee> | TaskCollabAssignee> =>
+                        // $FlowFixMe
```

Comment:
$FlowFixMe will be removed in a followup PR where the flow types get resolved @alexkrolick @jstoffan